pkg/logging/options.go

```diff
@@ -125,6 +136,21 @@ func DefaultCodeToLevel(logger log.Logger, code int) log.Logger {
 	return level.Error(logger)
 }
 
+func DefaultLevelledLogger(logger log.Logger, lvl string) log.Logger {
```

Comment:
Should we allow only `Info` and `Debug` in here as you mentioned in the PR description? Or perhaps we could validate level beforehand and only accept those two levels?

Reply:
See https://github.com/thanos-io/thanos/pull/4956#discussion_r770723183 but yes, or we could restrict the request logging config to only those two options while keeping the package more generic.

I didn't want to overhaul to much here without feedback from maintainers first :)